torchmetrics/metric.py

```diff
@@ -27,15 +26,17 @@
 
 from torchmetrics.utilities import apply_to_collection, rank_zero_warn
 from torchmetrics.utilities.data import _flatten, dim_zero_cat, dim_zero_mean, dim_zero_sum
+from torchmetrics.utilities.device_dtype_mixin import DeviceDtypeModuleMixin
 from torchmetrics.utilities.distributed import gather_all_tensors
+from torchmetrics.utilities.exceptions import MisconfigurationException
 from torchmetrics.utilities.imports import _LIGHTNING_AVAILABLE, _compare_version
 
 
 def jit_distributed_available() -> bool:
     return torch.distributed.is_available() and torch.distributed.is_initialized()
 
 
-class Metric(nn.Module, ABC):
+class Metric(DeviceDtypeModuleMixin, nn.Module, ABC):
```

Comment:
```suggestion
class Metric(DeviceDtypeModuleMixin, ABC):
```

the mixin already inherits from `nn.Module`

Reply:
Removed ! We can add it in another PR .